### PR TITLE
Proposed fix for #466: polynomial estimate_parameters

### PR DESCRIPTION
**Pull request by vidartf**
Proposed fix for #466. A little unsure about if these lines are correct when unfolding:
```python
if axis.index_in_array > 0:
    dc = np.rollaxis(dc, 1, 0)
...
if axis.index_in_array > 0:
    cmaps = np.rollaxis(cmaps, 0, 2)
```
@francisco-dlp you seem quite steady on the (un)folding code now. What do you think?

### TODO
 - [x] Confirm `rollaxis`
 - [x] Tests

### CODE REVIEW

File: hyperspy/_components/polynomial.py [hunk 1]
```diff
@@ -104,19 +104,23 @@ def estimate_parameters(self, signal, x1, x2, only_current=False):
...
-            if binned is True:
-                self.coefficients.map["values"] /= axis.scale
-            self.coefficients.map['is_set'][:] = True
-            signal.fold()
+            unfolded = signal.unfold()
+            try:
+                dc = signal.data
+                # For polyfit the spectrum goes in the first axis
+                if axis.index_in_array > 0:
+                    dc = np.rollaxis(dc, 1, 0)  # Unfolded, so use 1
```
Comment (francisco-dlp) [inline hyperspy/_components/polynomial.py]: > # Unfolded, so use 1

Good point, then why not simply `dc = dc.T`

File: hyperspy/_components/polynomial.py [hunk 2]
```diff
@@ -104,19 +104,23 @@ def estimate_parameters(self, signal, x1, x2, only_current=False):
...
+                cmaps = np.polyfit(axis.axis[i1:i2], dc[i1:i2, :],
+                                   self.get_polynomial_order())
+                if axis.index_in_array > 0:
+                    cmaps = np.rollaxis(cmaps, 0, 2)
```
Comment (francisco-dlp) [inline hyperspy/_components/polynomial.py]: Same here  `cmaps = cmaps.T` should be equivalent but more readable.

File: hyperspy/_components/polynomial.py (hunk 1 above)
Comment (vidartf) [inline hyperspy/_components/polynomial.py]: Missed that opportunity, thanks!

**Comment by vidartf** [comment]
Tests added, so if they look OK this should be ready for pull!

**Pull request merged**